modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/HeadphoneControlsSettingsFragment.kt

```diff
@@ -0,0 +1,169 @@
+import au.com.shiftyjelly.pocketcasts.compose.components.TextP50
+import au.com.shiftyjelly.pocketcasts.compose.theme
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.preferences.Settings.HeadphoneAction
+import au.com.shiftyjelly.pocketcasts.views.fragments.BaseFragment
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@AndroidEntryPoint
+class HeadphoneControlsSettingsFragment : BaseFragment() {
+    @Inject
+    lateinit var settings: Settings
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ) = ComposeView(requireContext()).apply {
+        setContent {
+            AppThemeWithBackground(theme.activeTheme) {
+                setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+                HeadphoneControlsSettingsPage(
+                    onBackPressed = {
+                        @Suppress("DEPRECATION")
+                        activity?.onBackPressed()
+                    }
+                )
+            }
+        }
+    }
+
+    @Composable
+    private fun HeadphoneControlsSettingsPage(
+        onBackPressed: () -> Unit,
+    ) {
+        Column {
+            ThemedTopAppBar(
+                title = stringResource(LR.string.settings_title_headphone_controls),
+                bottomShadow = true,
+                onNavigationClick = { onBackPressed() }
+            )
+            Column(
+                modifier = Modifier
+                    .verticalScroll(rememberScrollState())
+            ) {
+                TextP50(
+                    text = stringResource(LR.string.settings_headphone_controls_summary),
+                    color = MaterialTheme.theme.colors.primaryText02,
+                    modifier = Modifier.padding(16.dp)
+                )
+                PreviousActionRow(
+                    saved = settings.headphonePreviousActionFlow.collectAsState().value,
+                    onSave = {
+                        settings.setHeadphoneControlsPreviousAction(it)
+                    }
+                )
+                NextActionRow(
+                    saved = settings.headphoneNextActionFlow.collectAsState().value,
+                    onSave = {
+                        settings.setHeadphoneControlsNextAction(it)
+                    }
+                )
+                if (shouldShowConfirmationSoundRow()) {
+                    ConfirmationSoundRow(
+                        saved = settings.headphonePlayBookmarkConfirmationSoundFlow.collectAsState().value,
+                        onSave = {
+                            settings.setHeadphoneControlsPlayBookmarkConfirmationSound(it)
+                        }
+                    )
+                }
```

Comment:
Would it be good to move the state up so it's easier to add a preview to the component?

```suggestion
                val previousAction = settings.headphonePreviousActionFlow.collectAsState().value
                val nextAction = settings.headphoneNextActionFlow.collectAsState().value
                val confirmationSound = settings.headphonePlayBookmarkConfirmationSoundFlow.collectAsState().value

                HeadphoneControlsSettingsPage(
                    previousAction = previousAction,
                    nextAction = nextAction,
                    onPreviousActionSave = { settings.setHeadphoneControlsPreviousAction(it) },
                    onNextActionSave = { settings.setHeadphoneControlsNextAction(it) },
                    confirmationSound = confirmationSound,
                    onConfirmationSoundSave = { settings.setHeadphoneControlsPlayBookmarkConfirmationSound(it) },
                    onBackPressed = {
                        @Suppress("DEPRECATION")
                        activity?.onBackPressed()
                    }
                )
            }
        }
    }

    @Composable
    private fun HeadphoneControlsSettingsPage(
        previousAction: HeadphoneAction,
        nextAction: HeadphoneAction,
        onPreviousActionSave: (HeadphoneAction) -> Unit,
        onNextActionSave: (HeadphoneAction) -> Unit,
        confirmationSound: Boolean,
        onConfirmationSoundSave: (Boolean) -> Unit,
        onBackPressed: () -> Unit,
    ) {
        Column {
            ThemedTopAppBar(
                title = stringResource(LR.string.settings_title_headphone_controls),
                bottomShadow = true,
                onNavigationClick = { onBackPressed() }
            )
            Column(
                modifier = Modifier
                    .verticalScroll(rememberScrollState())
            ) {
                TextP50(
                    text = stringResource(LR.string.settings_headphone_controls_summary),
                    color = MaterialTheme.theme.colors.primaryText02,
                    modifier = Modifier.padding(16.dp)
                )
                PreviousActionRow(
                    saved = previousAction,
                    onSave = onPreviousActionSave
                )
                NextActionRow(
                    saved = nextAction,
                    onSave = onNextActionSave
                )
                if (previousAction == HeadphoneAction.ADD_BOOKMARK || nextAction == HeadphoneAction.ADD_BOOKMARK) {
                    ConfirmationSoundRow(
                        saved = confirmationSound,
                        onSave = onConfirmationSoundSave
                    )
                }
```

Reply:
I applied the changes and also added a preview in [a6b42ff](https://github.com/Automattic/pocket-casts-android/pull/1156/commits/a6b42ff993b26db56266af955645fcc672fc9736). Thanks for the suggestion. 🙏